### DISCUSSION TITLE
How to make a switched camera?

### DISCUSSION THREAD

**Discussion by the-brute-force** [BrenekH/kotlin-for-frc#135] [General]
Hi all, I'm moving from C++ and had some troubles porting some camera code. I'm trying to use `addSwitchedCamera(String name)` so I can set that as the camera on the DriverStation and just toggle through my USB cameras through `setSource()`.

The examples for the Java versions show the VideoSink being initialised with an implicitly, but when I try to use it in Kotlin, I get `Cannot access '<init>': it is protected in 'VideoSink'`

Here is what I'm trying to do with Java:
```java
UsbCamera Camera0;
UsbCamera Camera1;
VideoSink Server;

@Override
public void robotInit() {
    Camera0 = CameraServer.startAutomaticCapture(0);
    Camera1 = CameraServer.startAutomaticCapture(1);
    Server = CameraServer.addSwitchedCamera​("Switch");
    Server.setSource(Camera0);
}
```

Thanks in advanced.

**Comment by BrenekH** [comment]
This is the equivalent of the code you posted in Kotlin:

```kotlin
private lateinit var Camera0: UsbCamera
private lateinit var Camera1: UsbCamera
private lateinit var Server: VideoSink

override fun robotInit() {
    Camera0 = CameraServer.startAutomaticCapture(0)
    Camera1 = CameraServer.startAutomaticCapture(1)
    Server = CameraServer.addSwitchedCamera("Switch")
    Server.source = camera0
}
```

I couldn't tell you for certain why you were getting the error you posted without the code you tried, but I can tell you that it is raised if you try to instantiate the `VideoSink` class directly. This is because the constructor is marked as `protected` instead of `public`.

**Comment by the-brute-force** [comment]
That worked brilliantly, I was just trying to test example code.
